### PR TITLE
Move MeasurementGate, measure, and measure_each out of common_gates.py

### PR DESCRIPTION
**Pull request by Strilanc**
- The file has become too large, and we are splitting it up.

### CODE REVIEW

**Comment by mpharrigan** [comment]
xref https://github.com/quantumlib/Cirq/pull/2205#issuecomment-539260440

Can we keep compatibility imports in `common_gates` with a deprecation warning?

**Comment by Strilanc** [comment]
I'm not sure if it's possible to do that. We'd need new tooling at the very least. The issue is that these things are attributes at the module level, and we don't have a way to intercept the access. So we would effectively need to intercept usage after access, which would involve e.g. wrapping every element on the class.

I did originally want to prefix all the files with `_` to avoid this sort of thing.

File: cirq/ops/measure_util.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Copyright 2018 The Cirq Developers
```
Comment (dabacon) [inline cirq/ops/measure_util.py]: 2019

File: cirq/ops/measure_util_test.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Copyright 2018 The Cirq Developers
```
Comment (dabacon) [inline cirq/ops/measure_util_test.py]: 2019

**Comment by Strilanc** [comment]
This file is likely to grow beyond just `MeasurementGate`, e.g. measurement of pauli strings.

**Pull request merged**